### ISSUE TITLE
Failed to connect to PulseAudio

### ISSUE THREAD

**Issue by Vaelatern** [patroclos/PAmix#8]
This happens when pulseaudio is not running, but at the same time happens when pavucontrol is able to connect to pulse audio. I am happy to help debug exactly when this happens.

**Comment by patroclos** [comment]
So from your description is it right for me to assume that this does not always happen?
Also can you give me the following things:
- your pulseaudio version
- the content of the DISPLAY environment variable
- your /etc/pulse/daemon.conf
- were you using pamix over ssh

PS: thank you a lot for your contributions to this project!

**Comment by Vaelatern** [comment]
Correct, sometimes when I am playing media I am able to load pamix.

I never start pulse audio explicitely, I allow X to start it as needed.

```
$ pulseaudio --version
pulseaudio 9.0-rebootstrapped
$ echo $DISPLAY
:0
$ cat /etc/pulse/daemon.conf
# This file is part of PulseAudio.
#
# PulseAudio is free software; you can redistribute it and/or modify
# it under the terms of the GNU Lesser General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version.
#
# PulseAudio is distributed in the hope that it will be useful, but
# WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
# General Public License for more details.
#
# You should have received a copy of the GNU Lesser General Public License
# along with PulseAudio; if not, see <http://www.gnu.org/licenses/>.

## Configuration file for the PulseAudio daemon. See pulse-daemon.conf(5) for
## more information. Default values are commented out.  Use either ; or # for
## commenting.

; daemonize = no
; fail = yes
; allow-module-loading = yes
; allow-exit = yes
; use-pid-file = yes
; system-instance = no
; local-server-type = user
; enable-shm = yes
; shm-size-bytes = 0 # setting this 0 will use the system-default, usually 64 MiB
; lock-memory = no
; cpu-limit = no

; high-priority = yes
; nice-level = -11

; realtime-scheduling = yes
; realtime-priority = 5

; exit-idle-time = 20
; scache-idle-time = 20

; dl-search-path = (depends on architecture)

; load-default-script-file = yes
; default-script-file = /etc/pulse/default.pa

; log-target = auto
; log-level = notice
; log-meta = no
; log-time = no
; log-backtrace = 0

; resample-method = speex-float-1
; enable-remixing = yes
; enable-lfe-remixing = no
; lfe-crossover-freq = 0

; flat-volumes = yes

; rlimit-fsize = -1
; rlimit-data = -1
; rlimit-stack = -1
; rlimit-core = -1
; rlimit-as = -1
; rlimit-rss = -1
; rlimit-nproc = -1
; rlimit-nofile = 256
; rlimit-memlock = -1
; rlimit-locks = -1
; rlimit-sigpending = -1
; rlimit-msgqueue = -1
; rlimit-nice = 31
; rlimit-rtprio = 9
; rlimit-rttime = 200000

; default-sample-format = s16le
; default-sample-rate = 44100
; alternate-sample-rate = 48000
; default-sample-channels = 2
; default-channel-map = front-left,front-right

; default-fragments = 4
; default-fragment-size-msec = 25

; enable-deferred-volume = yes
; deferred-volume-safety-margin-usec = 8000
; deferred-volume-extra-delay-usec = 0
```

I am using pamix from the command line from a urxvt instance within the Awesome window manager, in Xorg.

**Comment by patroclos** [comment]
so if x starts pulseaudio as needed, is pulse getting killed when you are not playing anything?

**Comment by Vaelatern** [comment]
That sounds right. I'll do more rigorous testing and let you know.

**Comment by Vaelatern** [comment]
Whatever it is, pavucontrol does not appear affected.

**Comment by patroclos** [comment]
Interesting, so pulseaudio should be running.
Are you maybe running pamix or pavucontrol as a different user?

**Comment by Vaelatern** [comment]
No, always the same user.
Here is a recording. It turns out that running pavucontrol makes pamix run just fine.
https://asciinema.org/a/cpxdejltch7za8l1j0c5rzyf7

**Comment by patroclos** [comment]
Thats strange, can you check weather pulseaudio is actually running before you run pavucontrol and pamix fails?
Allthough pavucontrol should not be able to start a pulse daemon..
Also, did you do anything in pavucontrol?

**Comment by Vaelatern** [comment]
I moved the volume slider in pavucontrol in that instance.

**Comment by Vaelatern** [comment]
Will check back with another test

**Comment by Vaelatern** [comment]
When opening pavucontrol and doing nothing, then closing pavucontrol, pamix then works.

```
» pgrep pulse
» pamix      
Failed to connect to PulseAudio.
» pgrep pulse 
» pgrep pulse ; pamix
Failed to connect to PulseAudio.
» pgrep pulse ; pavucontrol # no pulse. pavucontrol works fine.
» pgrep pulse ; pavucontrol #pulse. pavucontrol works.
12641
» pgrep pulse ; pavucontrol #pulse. pavucontrol works.
12641
» pgrep pulse ; pamix  #pulse. pamix works.
12641
» 
```

While pavucontrol is up, pamix also works.

**Comment by patroclos** [comment]
ok, so your pavucontrol is starting a pulse daemon, pamix doesnt do that
you could try to change PA_CONTEXT_NOAUTOSPAWN to PA_CONTEXT_NOFLAGS in src/painterface.cpp , recompile and see if that works for you.

**Comment by Vaelatern** [comment]
It does. Now pamix successfully connects to pulse even when there is no daemon started, by starting a daemon.

**Comment by Vaelatern** [comment]
I have the patch and can PR if that works for you.

**Comment by patroclos** [comment]
Can now be configured in pamix.conf
Closes #12
